core/src/main/scala/kafka/log/remote/RemoteLogIndexEntry.scala

```diff
@@ -113,6 +117,60 @@ object RemoteLogIndexEntry {
     entry
   }
 
+  def readAll(is: InputStream): Seq[RemoteLogIndexEntry] = {
+    var index = Seq.empty[RemoteLogIndexEntry]
+
+    var done = false
+    while (!done) {
+      parseEntry(is) match {
+        case Some(entry) =>
+          index = index :+ entry
+        case _ =>
+          done = true
+      }
+    }
+
+    index
+  }
+
+  private def parseEntry(is: InputStream): Option[RemoteLogIndexEntry] = {
+    val magicAndEntryLengthBuffer = ByteBuffer.allocate(java.lang.Short.BYTES + java.lang.Short.BYTES)
+    Utils.readFully(is, magicAndEntryLengthBuffer)
+    if (magicAndEntryLengthBuffer.hasRemaining) {
+      None
+    } else {
+      magicAndEntryLengthBuffer.flip()
+      val magic = magicAndEntryLengthBuffer.getShort()
+      magic match {
+        case 0 =>
+          val entryLength = magicAndEntryLengthBuffer.getShort(java.lang.Short.BYTES)
```

Comment:
Should not `magicAndEntryLengthBuffer.getShort()` be sufficient as earlier `getShort` should have moved the position?

Reply:
Absolutely. Fixed.